packages/studio-base/src/panels/ThreeDeeRender/renderables/FoxgloveGrid.ts

```diff
@@ -42,6 +44,19 @@ const DEFAULT_MIN_COLOR = { r: 100 / 255, g: 47 / 255, b: 105 / 255, a: 1 };
 const DEFAULT_MAX_COLOR = { r: 227 / 255, g: 177 / 255, b: 135 / 255, a: 1 };
 const DEFAULT_RGB_BYTE_ORDER = "rgba";
 
+const COLOR_MODE_TO_GLSL: Record<string, number> = {
+  FLAT: 0,
+  RGB: 1,
+  RGBA: 2,
+  GRADIENT: 3,
+  COLORMAP: 4,
+};
+
+const COLOR_MAP_TO_GLSL: Record<string, number> = {
+  TURBO: 0,
+  RAINBOW: 1,
+};
```

Comment:
I'd prefix these keys just to aid clarity and avoid potential name conflicts with other shader defines.

```suggestion
const COLOR_MODE_TO_GLSL: Record<string, number> = {
  COLOR_MODE_FLAT: 0,
  COLOR_MODE_RGB: 1,
  COLOR_MODE_RGBA: 2,
  COLOR_MODE_GRADIENT: 3,
  COLOR_MODE_COLORMAP: 4,
};

const COLOR_MAP_TO_GLSL: Record<string, number> = {
  COLOR_MAP_TURBO: 0,
  COLOR_MAP_RAINBOW: 1,
};
```

Reply:
Furthermore if we are relying on the fact that each of these corresponds to one of `ColorModeSettings["colorMode"]`, then we should leverage the type system to enforce it. For example, ``const COLOR_MODE_TO_GLSL: { [K in ColorModeSettings["colorMode"] as `COLOR_MODE_${Uppercase<K>}`]: number }``

Reply:
yeah I wanted it to be type enforced but didn't know how to do it properly. Thanks!
good point on the collisions, will update

Reply:
:/ .toUpperCase doesn't keep the base string literal type. would it be alright to use the workaround outlined here to change the global string interface for uppercase and lowercase to maintain these types through this method? https://github.com/microsoft/TypeScript/issues/44268